### PR TITLE
core: several follow-ups for recent change for PrivateTmp=

### PR DESCRIPTION
**Pull request by yuwata**
Follow-ups for #32724.

### CODE REVIEW

File: src/core/execute-serialize.c [hunk 1]
```diff
@@ -2720,7 +2720,9 @@ static int exec_context_deserialize(ExecContext *c, FILE *f) {
                         if (r < 0)
                                 return r;
                 } else if ((val = startswith(l, "exec-context-private-tmp="))) {
-                        c->private_tmp = private_tmp_from_string(val);
+                        /* Before v257, the field was serialized with serialize_bool_elide(). Hence, "yes"
```
Comment (YHNdnzj) [inline src/core/execute-serialize.c]: No. There's no need. We pin the executor fd to guard against incompatible changes, which is completely different from how serialization of `Manager` and `Unit` objects are handled.

Comment (YHNdnzj) [reply src/core/execute-serialize.c]: `ExecContext` is _not_ serialized on reexec. We load the settings from unit fragments. This serialization format is solely used for sd-executor invocations.

Comment (yuwata) [reply src/core/execute-serialize.c]: Indeed. I found the comment in execute-serialize.h. Thanks.

Comment (yuwata) [reply src/core/execute-serialize.c]: The commit is dropped now.

**Review by YHNdnzj — APPROVED** [review]

**Comment by poettering** [comment]
oh, indeed!

**Pull request merged**